lib/graphql/scalar_type.rb

```diff
@@ -117,16 +117,14 @@ def raw_coercion_input(value)
         value.to_h
       elsif value.is_a?(Array)
         value.map { |element| raw_coercion_input(element) }
-      elsif value.is_a?(GraphQL::Language::Nodes::Enum)
```

Comment:
With this, enums will never be treated as a scalar anymore. Practically, only ID and String types are affected as the casting for numeric types should reject this